### ISSUE TITLE
Getting error `TypeError: WebDriver.__init__() got an unexpected keyword argument 'chrome_options'`

### ISSUE THREAD

**Issue by leockl** [alessandriniluca/postget#2]
**Reproducible Example**

```
from postget.Posts import Posts
from postget.exceptions.exceptions import *

def main():
    twitter_getter = Posts(username='MY-USERNAME', 
                           password='MY-PASSWORD', 
                           num_scrolls=100, 
                           query='(Tesla OR TSLA)', 
                           email_address='MY-EMAIL', 
                           mode=1, 
                           until='2023-01-01', 
                           since='2020-01-01', 
                           headless=True, 
                           chromedriver='MY-/FILE-/PATH-/TO-/CHROMEDRIVER-/FOLDER'
                           )
    try:
        twitter_getter.login()
    except ElementNotLoaded as e:
        raise e


    print("Setting query in the object")
    twitter_getter.set_query('test')

    print("Start Search, this will input the query and perform the scroll with the selected mode")
    try:
	    twitter_getter.search()
    except ElementNotLoaded as e:
	    raise e
    except NoTweetsReturned as e:
	    print(e)
	
    print("Printing returned results and going home")
    twitter_getter.print_results()
    twitter_getter.go_home()
    print("Clearing Results")
    twitter_getter.clear_tweets()
    print("quitting browser")
    twitter_getter.quit_browser()

if __name__=='__main__':
    main()
```

**Issue Description**
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[15], line 42
     39     twitter_getter.quit_browser()
     41 if __name__=='__main__':
---> 42     main()

Cell In[15], line 5, in main()
      4 def main():
----> 5     twitter_getter = Posts(username='MY-USERNAME', 
      6                            password='MY-PASSWORD', 
      7                            num_scrolls=100, 
      8                            query='(Tesla OR TSLA)', 
      9                            email_address='MY-EMAIL', 
     10                            mode=1, 
     11                            until='2023-01-01', 
     12                            since='2020-01-01', 
     13                            headless=True, 
     14                            chromedriver='[MY-/FILE-/PATH-/TO-/CHROMEDRIVER-/FOLDER](file:///MY-/FILE-/PATH-/TO-/CHROMEDRIVER-/FOLDER)'
     15                            )
     16     try:
     17         twitter_getter.login()

File [c:\users\leo_c\downloads\postget\postget\Posts.py:103](file:///C:/users/leo_c/downloads/postget/postget/Posts.py:103), in Posts.__init__(self, username, password, query, email_address, wait_scroll_base, wait_scroll_epsilon, num_scrolls, mode, since_id, max_id, since, until, since_time, until_time, headless, chromedriver, root)
    101     print('[postget]: Running in root mode. This is not recommended for security reasons, disabling sandbox to allow run chromium.')
    102 if chromedriver != 'none':
--> 103     self.driver=webdriver.Chrome(chromedriver, chrome_options=self.chrome_options)
    104 else:
    105     self.driver=webdriver.Chrome(shutil.which('chromedriver'), chrome_options=self.chrome_options)

TypeError: WebDriver.__init__() got an unexpected keyword argument 'chrome_options'
```

**Expected Behavior**
Expected the script to run without any errors.

**Installed Versions**
postget-1.1.1
Chromedriver 113.0.5672.63
Chromium 117.0.5904.0
Python 3.10.9
Conda 23.3.1
Windows 11

**Extra Discription**
Based on the Selenium documentation [here](https://www.selenium.dev/documentation/webdriver/drivers/options/), the Python code version of the 'chrome options' parameter in `selenium.webdriver.Chrome()` appears to be called `options` rather than `chrome_options`.

**Comment by alessandriniluca** [comment]
First of all, thanks for pointing it out!

I've done some tests, with both python 3.10 and 3.11:
I've tried now for the first time postget on windows, and it seems working correctly under these conditions:

- Just give the google chrome's path to postget, instead of chromium and chromedriver. My chrome version is: `115.0.5790.110`, you should find the executable at `C:\Program Files\Google\Chrome\Application\chrome.exe`  (or something equivalent according to where you installed it)
- run it in **non** headless mode (parameter `headless` needs to be set to `False`). This seems not being a windows-related problem, the same seems happening now on Linux (this needs some investigation, probably something changed from the first version of the code)
- Set chrome with dark theme (it should be enough to enable that option in the browser's settings)

Let me know if the problem persists, in case this did not solve the problem I'll upate the code.
If with these changes is working also for you, I'll update the readme file to add documntation also for windows.

**Comment by alessandriniluca** [comment]
First of all, thanks for pointing it out! I've made a quick update that should be fixing that, if `chrome_options` is the problem. Unfortunately at the moment I cannot test if it is working, since that problem does not appears to me, hence I don't know if the problem persists (for me, both `chrome_options` and `options` works). I'm making a quick push of the changes on the branch `feature/windows_support`, I would kindly ask you to let me know if this solved the problem, otherwise further investigation is needed.
By the way, **I noticed that the headless mode right now could cause some problems, and return nothing**, but this seems not being a windows-related problem, since this started happening also on linux. Probably something in the website changed, hence for now run it in **non headless mode** (`headless=False`)

**Comment by leockl** [comment]
To ensure I start things from fresh, I deleted all the existing postget files in my local device and re-cloned your repo into my local device laptop. I re-ran the script but its still giving me that same error:

 **Reproducible Example**
```
from postget.Posts import Posts
from postget.exceptions.exceptions import *

def main():
    twitter_getter = Posts(username='wishhokaido', 
                           password='Twitapp08', 
                           num_scrolls=100, 
                           query='(Tesla OR TSLA)', 
                           email_address='wishshibuya@outlook.com', 
                           mode=1, 
                           until='2023-01-14', 
                           since='2023-01-01', 
                           headless=False,  # changed to False
                           chromedriver='C:/Users/leo_c/Downloads/chromedriver_win32'
                           )
    try:
        twitter_getter.login()
    except ElementNotLoaded as e:
        raise e


    print("Setting query in the object")
    twitter_getter.set_query('test')

    print("Start Search, this will input the query and perform the scroll with the selected mode")
    try:
	    twitter_getter.search()
    except ElementNotLoaded as e:
	    raise e
    except NoTweetsReturned as e:
	    print(e)
	
    print("Printing returned results and going home")
    twitter_getter.print_results()
    twitter_getter.go_home()
    print("Clearing Results")
    twitter_getter.clear_tweets()
    print("quitting browser")
    twitter_getter.quit_browser()

if __name__=='__main__':
    main()
```

**Issue Description**
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[2], line 42
     39     twitter_getter.quit_browser()
     41 if __name__=='__main__':
---> 42     main()

Cell In[2], line 5, in main()
      4 def main():
----> 5     twitter_getter = Posts(username='wishhokaido', 
      6                            password='Twitapp08', 
      7                            num_scrolls=100, 
      8                            query='(Tesla OR TSLA)', 
      9                            email_address='wishshibuya@outlook.com', 
     10                            mode=1, 
     11                            until='2023-01-14', 
     12                            since='2023-01-01', 
     13                            headless=False,  # changed to False
     14                            chromedriver='[C:/Users/leo_c/Downloads/chromedriver_win32](file:///C:/Users/leo_c/Downloads/chromedriver_win32)'
     15                            )
     16     try:
     17         twitter_getter.login()

File [c:\users\leo_c\downloads\postget\postget\Posts.py:103](file:///C:/users/leo_c/downloads/postget/postget/Posts.py:103), in Posts.__init__(self, username, password, query, email_address, wait_scroll_base, wait_scroll_epsilon, num_scrolls, mode, since_id, max_id, since, until, since_time, until_time, headless, chromedriver, root)
    101     print('[postget]: Running in root mode. This is not recommended for security reasons, disabling sandbox to allow run chromium.')
    102 if chromedriver != 'none':
--> 103     self.driver=webdriver.Chrome(chromedriver, chrome_options=self.chrome_options)
    104 else:
    105     self.driver=webdriver.Chrome(shutil.which('chromedriver'), chrome_options=self.chrome_options)

TypeError: WebDriver.__init__() got an unexpected keyword argument 'chrome_options'
```

**Comment by alessandriniluca** [comment]
It seems that you do not have the changes I've done to the code: for example, at line 103 the issue says that your code is:
`self.driver=webdriver.Chrome(chromedriver, chrome_options=self.chrome_options)`
Instead, that same line in this branch (`feature/windows_support`) you can find:
`self.driver=webdriver.Chrome(chromedriver, options=self.chrome_options)`.
If you notice, the new code in this branch should be using `options`, which is the parameter that also the [example](https://www.selenium.dev/documentation/webdriver/drivers/options/#normal-default) in the official guide that you pointed me out is using for python.
I think that this could be solved by:

- Open in terminal (or with [git bash](https://git-scm.com/downloads)) the folder where you have done the pull of the repository
- run `git pull` to update and download all the changes
- run `git switch feature/windows_support`: this is necessary, since the changes I've made are not yet merged in the `main` or in `develop` branch yet, since other issues may arise with windows
- Only after having switched to the branch, install the package from the source code.

Let me know if this is working!

**Comment by leockl** [comment]
For precaution, I deleted all existing postget files in my local device and re-cloned your repo into my local device laptop. I am still getting the same error:

**Reproducible Example**
```
from postget.Posts import Posts
from postget.exceptions.exceptions import *

def main():
    twitter_getter = Posts(username='MY-USERNAME', 
                           password='MY-PASSWORD', 
                           num_scrolls=100, 
                           query='(Tesla OR TSLA)', 
                           email_address='MY-EMAIL', 
                           mode=1, 
                           until='2023-01-14', 
                           since='2023-01-01', 
                           headless=False,  # changed to False
                           chromedriver='C:/Users/leo_c/Downloads/chromedriver_win32'
                           )
    try:
        twitter_getter.login()
    except ElementNotLoaded as e:
        raise e


    print("Setting query in the object")
    twitter_getter.set_query('test')

    print("Start Search, this will input the query and perform the scroll with the selected mode")
    try:
	    twitter_getter.search()
    except ElementNotLoaded as e:
	    raise e
    except NoTweetsReturned as e:
	    print(e)
	
    print("Printing returned results and going home")
    twitter_getter.print_results()
    twitter_getter.go_home()
    print("Clearing Results")
    twitter_getter.clear_tweets()
    print("quitting browser")
    twitter_getter.quit_browser()

if __name__=='__main__':
    main()
```

**Issue Description**
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[2], line 42
     39     twitter_getter.quit_browser()
     41 if __name__=='__main__':
---> 42     main()

Cell In[2], line 5, in main()
      4 def main():
----> 5     twitter_getter = Posts(username='MY-USERNAME', 
      6                            password='MY-PASSWORD', 
      7                            num_scrolls=100, 
      8                            query='(Tesla OR TSLA)', 
      9                            email_address='MY-EMAIL', 
     10                            mode=1, 
     11                            until='2023-01-14', 
     12                            since='2023-01-01', 
     13                            headless=False,  # changed to False
     14                            chromedriver='[C:/Users/leo_c/Downloads/chromedriver_win32](file:///C:/Users/leo_c/Downloads/chromedriver_win32)'
     15                            )
     16     try:
     17         twitter_getter.login()

File [c:\users\leo_c\downloads\postget\postget\Posts.py:103](file:///C:/users/leo_c/downloads/postget/postget/Posts.py:103), in Posts.__init__(self, username, password, query, email_address, wait_scroll_base, wait_scroll_epsilon, num_scrolls, mode, since_id, max_id, since, until, since_time, until_time, headless, chromedriver, root)
    101     print('[postget]: Running in root mode. This is not recommended for security reasons, disabling sandbox to allow run chromium.')
    102 if chromedriver != 'none':
--> 103     self.driver=webdriver.Chrome(chromedriver, chrome_options=self.chrome_options)
    104 else:
    105     self.driver=webdriver.Chrome(shutil.which('chromedriver'), chrome_options=self.chrome_options)

TypeError: WebDriver.__init__() got an unexpected keyword argument 'chrome_options'
```

**Comment by alessandriniluca** [comment]
It seems that you do not have the changes I've done to the code: for example, at line 103 the issue says that your code is:
`self.driver=webdriver.Chrome(chromedriver, chrome_options=self.chrome_options)`
Instead, that same line in this branch (`feature/windows_support`) you can find:
`self.driver=webdriver.Chrome(chromedriver, options=self.chrome_options)`.
If you notice, the new code in this branch should be using `options`, which is the parameter that also the [example](https://www.selenium.dev/documentation/webdriver/drivers/options/#normal-default) in the official guide that you pointed me out is using for python.
I think that this could be solved by:

- Open in terminal (or with [git bash](https://git-scm.com/downloads)) the folder where you have done the pull of the repository
- run `git pull` to update and download all the changes
- run `git switch feature/windows_support`: this is necessary, since the changes I've made are not yet merged in the `main` or in `develop` branch yet, since other issues may arise with windows
- Only after having switched to the branch, install the package from the source code.

Let me know if this is working!

**Comment by leockl** [comment]
Sorry for not doing the branch change. Didn't know about this, but now I do!

Made this change and I am still getting the same error:
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[1], line 42
     39     twitter_getter.quit_browser()
     41 if __name__=='__main__':
---> 42     main()

Cell In[1], line 5, in main()
      4 def main():
----> 5     twitter_getter = Posts(username='MY-USERNAME', 
      6                            password='MY-PASSWORD', 
      7                            num_scrolls=100, 
      8                            query='(Tesla OR TSLA)', 
      9                            email_address='MY-EMAIL', 
     10                            mode=1, 
     11                            until='2023-01-14', 
     12                            since='2023-01-01', 
     13                            headless=False,  # changed to False
     14                            chromedriver='[C:/Users/leo_c/Downloads/chromedriver_win32](file:///C:/Users/leo_c/Downloads/chromedriver_win32)'
     15                            )
     16     try:
     17         twitter_getter.login()

File [c:\users\leo_c\downloads\postget\postget\Posts.py:102](file:///C:/users/leo_c/downloads/postget/postget/Posts.py:102), in Posts.__init__(self, username, password, query, email_address, wait_scroll_base, wait_scroll_epsilon, num_scrolls, mode, since_id, max_id, since, until, since_time, until_time, headless, chromedriver, root)
    100     print('[postget]: Running in root mode. This is not recommended for security reasons, disabling sandbox to allow run chromium.')
    101 if chromedriver != 'none':
--> 102     self.driver=webdriver.Chrome(chromedriver, options=self.chrome_options)
    103 else:
    104     self.driver=webdriver.Chrome(shutil.which('chromedriver'), options=self.chrome_options)

TypeError: WebDriver.__init__() got multiple values for argument 'options'
```

**Comment by alessandriniluca** [comment]
No problem for the branch! While checking the new error, I found [this issue on stack overflow](https://stackoverflow.com/questions/76428561/typeerror-webdriver-init-got-multiple-values-for-argument-options). This is caused by selenium starting from version `4.10.0`. By testing it, to solve this problem is sufficient to downgrade selenium to `4.9.0`.
In the meanwhile I'll close this issue to let people know that, for now, it is sufficient to downgrade selenium if they are encountering this error.
If someone needs a new version of the code in order to work also with newer versions of selenium for whatever reason, just open a new issue to request this, and I'll work on it!
